sdk/tests/integration/ClientIntegrationTest.cc

```diff
@@ -124,52 +130,36 @@ TEST_F(ClientIntegrationTest, ConnectToLocalNode)
-  Client client = Client::forNetwork(networkMap);
+  const AccountId accountId_3 = AccountId::fromString("0.0.3");
+  const AccountId accountId_4 = AccountId::fromString("0.0.4");
+  const AccountId accountId_5 = AccountId::fromString("0.0.5");
+  const AccountId accountId_6 = AccountId::fromString("0.0.6");
+  const AccountId accountId_7 = AccountId::fromString("0.0.7");
+
+  AccountBalance accountBalance_3;
+  AccountBalance accountBalance_4;
+  AccountBalance accountBalance_5;
+  AccountBalance accountBalance_6;
+  AccountBalance accountBalance_7;
 
-  // When / Then
-  EXPECT_THROW(client.setMinBackoff(getNegativeBackoffTime()), std::invalid_argument); // INVALID_ARGUMENT
-  EXPECT_THROW(client.setMinBackoff(getAboveMaxBackoffTime()), std::invalid_argument); // INVALID_ARGUMENT
-}
-
-//-----
-TEST_F(ClientIntegrationTest, SetValidMinBackoff)
-{
-  // Given
   std::unordered_map<std::string, AccountId> networkMap;
-  Client client = Client::forNetwork(networkMap);
-
-  // When / Then
-  EXPECT_NO_THROW(client.setMinBackoff(getZeroBackoffTime()));
-  EXPECT_NO_THROW(client.setMinBackoff(DEFAULT_MIN_BACKOFF));
-  EXPECT_NO_THROW(client.setMinBackoff(DEFAULT_MAX_BACKOFF));
-}
+  networkMap.insert(std::pair<std::string, AccountId>("34.94.106.61:50211", accountId_3));
+  networkMap.insert(std::pair<std::string, AccountId>("35.237.119.55:50211", accountId_4));
+  networkMap.insert(std::pair<std::string, AccountId>("35.245.27.193:50211", accountId_5));
 
-//-----
-TEST_F(ClientIntegrationTest, SetInvalidMaxBackoff)
-{
-  // Given
-  std::unordered_map<std::string, AccountId> networkMap;
   Client client = Client::forNetwork(networkMap);
 
   // When / Then
-  EXPECT_THROW(client.setMaxBackoff(getNegativeBackoffTime()), std::invalid_argument); // INVALID_ARGUMENT
-  EXPECT_THROW(client.setMaxBackoff(getZeroBackoffTime()), std::invalid_argument);     // INVALID_ARGUMENT
-  EXPECT_THROW(client.setMaxBackoff(getBelowMinBackoffTime()), std::invalid_argument); // INVALID_ARGUMENT
-  EXPECT_THROW(client.setMaxBackoff(getAboveMaxBackoffTime()), std::invalid_argument); // INVALID_ARGUMENT
-}
+  std::unordered_map<std::string, AccountId> newNetworkMap;
+  newNetworkMap.insert(std::pair<std::string, AccountId>("35.237.119.55:50211", accountId_6));
+  newNetworkMap.insert(std::pair<std::string, AccountId>("35.245.27.193:50211", accountId_7));
 
-//-----
-TEST_F(ClientIntegrationTest, SetValidMaxBackoff)
-{
-  // Given
-  std::unordered_map<std::string, AccountId> networkMap;
-  Client client = Client::forNetwork(networkMap);
+  client.setNetwork(newNetworkMap);
 
   // When / Then
-  EXPECT_NO_THROW(client.setMaxBackoff(DEFAULT_MIN_BACKOFF));
-  EXPECT_NO_THROW(client.setMaxBackoff(DEFAULT_MAX_BACKOFF));
-}
\ No newline at end of file
+  ASSERT_NO_THROW(accountBalance_6 = AccountBalanceQuery().setAccountId(accountId_6).execute(client));
+  ASSERT_NO_THROW(accountBalance_7 = AccountBalanceQuery().setAccountId(accountId_7).execute(client));
+}
```

Comment:
This is fine for now but I think we should really try to enforce going forward that the integration tests do not communicate with previewnet/testnet/mainnet nodes, hence my previous comments about waiting for the multi-node setup to be complete. A user should be able to test locally with that set-up if they so desire, and seeing as how this test contains URLs for testnet consensus nodes, these tests would not pass.

Reply:
I totally agree with you - using hardcoded URLs in the integration tests is a bad practice.
Integration tests should communicate only with the local node and this approach should be used in all of the Hedera's SDKs.
For now I will suggest to add a new JSON config files containing all of the consensus nodes for mainnet / previewnet / test. I will track this change in https://github.com/hashgraph/hedera-sdk-cpp/issues/525.

Reply:
I don't think this is necessary as we should not be communicating with testnet nodes to begin with. We would add this JSON file ultimately to remove it when we get the multi-node setup integrated. I think we should leave it for now and then go through this again when we get the multi-node setup working.